registry/servicediscovery/store/cache_manager.go

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package store
+
+import (
+	"encoding/gob"
+	"os"
+	"sync"
+	"time"
+)
+
+import (
+	"github.com/dubbogo/gost/log/logger"
+
+	"github.com/hashicorp/golang-lru"
+)
+
+type CacheManager struct {
+	name         string        // The name of the cache manager
+	cacheFile    string        // The file path where the cache is stored
+	cacheExpired time.Duration // The duration after which the cache expires
+	stop         chan struct{} // Channel used to stop the cache expiration routine
+	cache        *lru.Cache    // The LRU cache implementation
+	lock         sync.Mutex
+}
+
+type Item struct {
+	Key   string
+	Value interface{}
+}
+
+// NewCacheManager creates a new CacheManager instance.
+// It initializes the cache manager with the provided parameters and starts a routine for cache expiration.
+func NewCacheManager(name, cacheFile string, cacheExpired time.Duration, maxCacheSize int) (*CacheManager, error) {
+	cm := &CacheManager{
+		name:         name,
+		cacheFile:    cacheFile,
+		cacheExpired: cacheExpired,
+		stop:         make(chan struct{}),
+	}
+	cm.cache, _ = lru.New(maxCacheSize)
+
+	// Check if the cache file exists and load the cache if it does
+	if _, err := os.Stat(cacheFile); err == nil {
+		if err = cm.loadCache(); err != nil {
+			logger.Warnf("Failed to load the cache file:[%s].The err is %v", cm.cacheFile, err)
+		}
+	}
+	go cm.RunDumpTask()
+
+	return cm, nil
+}
+
+// Get retrieves the value associated with the given key from the cache.
+func (cm *CacheManager) Get(key string) (interface{}, bool) {
+	return cm.cache.Get(key)
+}
+
+// Set sets the value associated with the given key in the cache.
+func (cm *CacheManager) Set(key string, value interface{}) {
+	cm.cache.Add(key, value)
+}
+
+// Delete removes the value associated with the given key from the cache.
+func (cm *CacheManager) Delete(key string) {
+	cm.cache.Remove(key)
+}
```

Comment:
Is there a data race if we don't protect reading/writing `cm.cache` by the lock? Further, if dumping from cache is in process and no lock protects `cm.cache`, it might have some dirty data.